### PR TITLE
cloud_functions: alarm chains falling behind

### PR DESCRIPTION
**Pull request by panoel**
This PR Checks the latest observed block time on each chain (as collected by the watcher) and alarms (via slackbot) those chains that have fallen behind >= 24 hours.  It will only alarm a chain once per 24 hours.

### CODE REVIEW

File: cloud_functions/src/alarmMissingVaas.ts [hunk 1]
```diff
@@ -246,6 +249,90 @@ async function getLastBlockTimeFromFirestore(): Promise<LatestTimeByChain> {
...
 }
 
+async function alarmOldBlockTimes(latestTimes: LatestTimeByChain): Promise<void> {
+  let alarmsToStore: AlarmedChainTime[] = [];
+  // Read in the already alarmed chains.
+  const alarmedChains: Map<ChainId, AlarmedChainTime> = await getAlarmedChainsFromFirestore();
+  if (alarmedChains && alarmedChains.size > 0) {
+    alarmedChains.forEach((alarm) => {
+      alarmsToStore.push(alarm);
+    });
```
Comment (barnjamin) [inline cloud_functions/src/alarmMissingVaas.ts:259]: I think you can do this with a spread operator? like ``alarmsToStore = [...alarmedChains]`

File: cloud_functions/src/alarmMissingVaas.ts [hunk 2]
```diff
@@ -246,6 +249,90 @@ async function getLastBlockTimeFromFirestore(): Promise<LatestTimeByChain> {
...
+  } else {
+    console.log('no alarmed chains found in firestore');
+  }
+  // Walk all chains and check the latest block time.
+  const now = new Date();
+  for (const chain of Object.keys(latestTimes)) {
+    const chainId = chain as any as ChainId;
+    const latestTime = latestTimes[chainId]?.latestTime;
+    if (latestTime) {
```
Comment (barnjamin) [inline cloud_functions/src/alarmMissingVaas.ts:268]: nit: can you swap this to `!latestTime` and continue?

**Review by barnjamin — APPROVED** [review]

File: cloud_functions/src/alarmMissingVaas.ts (hunk 2 above)
Comment (panoel) [reply cloud_functions/src/alarmMissingVaas.ts:268]: Changed.

File: cloud_functions/src/alarmMissingVaas.ts (hunk 1 above)
Comment (panoel) [reply cloud_functions/src/alarmMissingVaas.ts:259]: Changed

**Review by barnjamin — APPROVED** [review]

**Pull request merged**